marimo/_runtime/packages/pypi_package_manager.py

```diff
@@ -48,6 +50,42 @@ class UvPackageManager(PypiPackageManager):
     async def _install(self, package: str) -> bool:
         return self.run(["uv", "pip", "install", package])
 
+    def update_notebook_script_metadata(
+        self,
+        filepath: str,
+        import_namespaces_to_add: List[str],
+        import_namespaces_to_remove: List[str],
+    ) -> None:
+        # Convert from module name to package name
+        packages_to_add = [
+            self.module_to_package(im) for im in import_namespaces_to_add
+        ]
+        packages_to_remove = [
+            self.module_to_package(im) for im in import_namespaces_to_remove
+        ]
+
+        # Filter to packages that are found by "uv pip show"
+        packages_to_add = [
+            im for im in packages_to_add if self._is_installed(im)
+        ]
+        packages_to_remove = [
+            im for im in packages_to_remove if self._is_installed(im)
+        ]
+
+        if packages_to_add:
+            self.run(
+                ["uv", "--quiet", "add", "--script", filepath]
+                + packages_to_add
+            )
+        if packages_to_remove:
+            self.run(
+                ["uv", "--quiet", "remove", "--script", filepath]
+                + packages_to_remove
+            )
```

Comment:
@mscolnick `uv remove` errors out if you try to remove a package that isn't installed, which is why I also filter `packages_to_remove`. An alternative would be to try to remove the packages one by one instead of all at once